src/Connection/Internal/Http2ConnectionProcessor.php

```diff
@@ -419,6 +416,31 @@ public function handleHeaders(int $streamId, array $pseudo, array $headers): voi
             }
         });
 
+        $response = new Response(
+            '2',
+            $status,
+            Status::getReason($status),
+            $headers,
+            new InMemoryStream,
+            $stream->request
+        );
+
+        if ($status < 200) {
+            $onInformationalResponse = $stream->request->getInformationalResponseHandler();
+
+            if ($onInformationalResponse !== null) {
+                asyncCall(function () use ($onInformationalResponse, $response, $streamId) {
+                    try {
+                        yield call($onInformationalResponse, $response);
+                    } catch (\Throwable $e) {
+                        $this->handleStreamException(new Http2StreamException('Informational response handler threw an exception', $streamId, self::CANCEL));
+                    }
+                });
+            }
```

Comment:
If we do it async here, we should probably do the same for HTTP/1?

Reply:
I've no opinion on that: I just borrowed from the way event listener are called. What would you advise?